persist/fs/manager.go

```diff
@@ -82,21 +82,21 @@ func (pm *persistManager) persist(
 	rateLimitMbps := pm.rateLimitOpts.LimitMbps()
 	if pm.rateLimitOpts.LimitEnabled() && rateLimitMbps > 0.0 {
 		now := pm.nowFn()
-		if pm.lastCheck.IsZero() {
+		if pm.start.IsZero() {
 			pm.start = now
-			pm.lastCheck = now
-		} else if now.Sub(pm.lastCheck) >= pm.rateLimitOpts.LimitCheckInterval() {
-			pm.lastCheck = now
+		} else if pm.count == pm.rateLimitOpts.LimitCheckEvery() {
```

Comment:
Is it safer perhaps to `pm.count %  pm.rateLimitOpts.LimitCheckEvery() == 0` and only reset to zero on `reset()` call?